system/lib/wasmfs/syscalls.cpp

```diff
@@ -1301,6 +1301,86 @@ int __syscall_fallocate(int fd, int mode, uint64_t off, uint64_t len) {
   return 0;
 }
 
+int __syscall_fcntl64(int fd, int cmd, ...) {
+  auto fileTable = wasmFS.getFileTable().locked();
+  auto openFile = fileTable.getEntry(fd);
+  if (!openFile) {
+    return -EBADF;
+  }
+
+  switch (cmd) {
+    case F_DUPFD: {
+      int newfd;
+      va_list v1;
+      va_start(v1, cmd);
+      newfd = va_arg(v1, int);
+      va_end(v1);
+      if (newfd < 0) {
+        return -EINVAL;
+      }
+
+      // Find the first available fd at arg or after.
+      // TODO: Should we check for a limit on the max FD number, if we have one?
+      while (1) {
+        if (!fileTable.getEntry(newfd)) {
+          fileTable.setEntry(newfd, openFile);
+          return newfd;
+        }
+        newfd++;
+      }
+    }
+    case F_GETFD:
+    case F_SETFD:
+      // FD_CLOEXEC makes no sense for a single process.
+      return 0;
+    case F_GETFL:
+      return openFile->locked().getFlags();
+    case F_SETFL: {
+      int flags;
+      va_list v1;
+      va_start(v1, cmd);
+      flags = va_arg(v1, int);
+      va_end(v1);
+      // TODO: check validity
+      openFile->locked().setFlags(flags);
```

Comment:
I guess we could do that too. It would be a divergence from the old FS, though, so there are benefits both ways.

Reply:
I would be surprised if anyone depended on Emscripten being able to change flags that Linux doesn't support changing. Do we have tests that depend on this? If not, I think it would be better to try to match Linux.

Reply:
Fair points, changed to error and to match linux more.

It seems like the syscall allows changing `O_APPEND` which we do support already. Looks like that is what is in the test actually. So this PR will now ignore what it should, and error on anything but that one flag.